client/reader/search-stream/search-follow-button.jsx

```diff
@@ -44,22 +47,55 @@ class SearchFollowButton extends Component {
 				);
 			}
 		}
+		return isPotentialFeedUrl;
+	};
 
-		// If not a potential feed then don't show the follow button
-		if ( ! isPotentialFeedUrl ) {
+	render() {
+		const { query, translate, feeds } = this.props;
+
+		// If the search query hasn't found a feed and the query doesn't look like a feed URL then don't show the follow button
+		if ( ! feeds && ! this.isPotentialFeedUrl( query ) ) {
 			return null;
 		}
 
-		let isFollowing = false;
-		let followTitle = withoutHttp( query );
+		// Check that the query is a URL
+		// Then Loop through feeds and find the feed URL that contains the query
+		// If we find a feed then set the feed object
+		let feed;
+		if ( resemblesUrl( query ) ) {
+			feed = feeds?.find( ( f ) => f.feed_URL.includes( withoutHttp( query ) ) );
+		}
 
-		if ( feed && feed.name?.length ) {
-			isFollowing = feed.is_following;
-			followTitle = feed.name;
+		// If no feed found, then don't show the follow button
+		if ( ! feed ) {
+			return null;
 		}
 
+		// We can use the feed to create a follow button
+
 		// If already following this feed then don't show the follow button
-		if ( isFollowing ) {
+		if ( feed.is_following !== undefined && feed.is_following === true ) {
```

Comment:
I think we should remove this logic. As it currently stands, if you search for a feed that you are already following, you will see neither the site result with the follow button nor the normal search results.